### PR TITLE
Drifted rlp cinsistency

### PR DESCRIPTION
**Pull request by CeciliaZ030**
### Description

Check the rlp encoded (key, value) pair of a drifted node is consistent with the original. 

### Type of change

- [x] Require that len(drifted pair) = len(drifted key) + len(original value)

### CODE REVIEW

File: zkevm-circuits/src/mpt_circuit/account_leaf.rs [hunk 1]
```diff
@@ -180,7 +181,10 @@ impl<F: Field> AccountLeafConfig<F> {
                     require!((1, leaf_rlc, rlp_key.rlp_list.num_bytes(), config.parent_data[is_s.idx()].rlc) => @KECCAK);
                 }}
 
+                // Cecilia: here copy
+
```
Comment (Brechtpd) [inline zkevm-circuits/src/mpt_circuit/account_leaf.rs]: remove

File: zkevm-circuits/src/mpt_circuit/account_leaf.rs [hunk 2]
```diff
@@ -180,7 +181,10 @@ impl<F: Field> AccountLeafConfig<F> {
...
                 // Check the RLP encoding consistency.
+                //               node      l[0],value_rlp_bytes[0] <-(l[1] = [n, b, s, c])
```
Comment (Brechtpd) [inline zkevm-circuits/src/mpt_circuit/account_leaf.rs]: remove

**Review by Brechtpd — APPROVED** [review]

File: zkevm-circuits/src/mpt_circuit/account_leaf.rs [hunk 3]
```diff
@@ -180,7 +181,10 @@ impl<F: Field> AccountLeafConfig<F> {
                     require!((1, leaf_rlc, rlp_key.rlp_list.num_bytes(), config.parent_data[is_s.idx()].rlc) => @KECCAK);
                 }}
 
+                // Cecilia: here copy
```
Comment (CeciliaZ030) [inline zkevm-circuits/src/mpt_circuit/account_leaf.rs]:
```suggestion
```

File: zkevm-circuits/src/mpt_circuit/account_leaf.rs (hunk 2 above)
Comment (CeciliaZ030) [inline zkevm-circuits/src/mpt_circuit/account_leaf.rs]:
```suggestion
```

**Pull request merged**